sycl/include/sycl/detail/device_filter.hpp

```diff
@@ -19,6 +19,51 @@ namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace detail {
 
+// ---------------------------------------
+// ONEAPI_DEVICE_SELECTOR support
+
+// the ONEAPI_DEVICE_SELECTOR string gets broken down into these targets
+// will will match devices.
+struct ods_target {
+public:
+  bool HasBackend = false;
+  backend Backend = backend::all;
```

Comment:
Would be nice to use `std::optional` here instead, but I suppose this is done to make it work with the same as `device_filter`? How much work would it be to change it both places?

Reply:
an excellent suggestion. done.